### ISSUE TITLE
Retry request in case of failure

### ISSUE THREAD

**Issue by GeorgFleig** [TwinProduction/gatus#59]
It would be great if Gatus had the possibility to retry a failed request `X` times with `Y` interval to verify an endpoint is "really" down and it is not just a short hiccup.

I saw this is already implemented for alerts (`services[].alerts[].failure-threshold`), but with this setting it will be shown as down on the dashboard anyway. And the interval for the retry is the same.

I was thinking of something like this:
```
  - name: google
    url: "https://google.com"
    interval: 1m
    retry-count: 2
    retry-interval: 10s
    conditions:
      - "[STATUS] == 200"
```
In case `retry-interval` is not set, `interval` is used. Not sure if it makes sense though to have the alert failure threshold and a generic retry logic.

**Comment by TwinProduction** [comment]
IMHO, if you need to retry the same request, it means the service is, at the very least, unstable, in which case, the dashboard should reflect that in one way or another.

Perhaps a new yellow icon with a `~` could be used to visually represent that a service's health check has passed, but only after one or more retry, but showing a green dashboard when a retry was required seems very counter intuitive.

**Comment by TwinProduction** [comment]
Closing this issue for the aforementioned reason.

**Comment by davidpanic** [comment]
Yes, this is slightly counterintuitive, but for a lot of use cases it makes sense, for example Kubernetes deployments with a restart strategy that are expected to have a short downtime from time to time, that doesn't necessarily indicate an error or cause for panic. If the user is knowingly setting a retry on a health check I think it makes sense to allow them to do that.
Please reconsider this 😄

**Comment by TwiN** [comment]
If your Kubernetes deployments cause downtime, I'd say that's a sign of instability as well, so it does indicate an error 😆

That said, If you want, you can create a new issue/feature request (make sure to refer this one), and if people show enough interest in the feature, I may reconsider my stance.

**Comment by filips123** [comment]
One use-case for this is if the base interval is slower, but you want to check the service more often when you detect it's down. So, if you have a 10 minute interval, one short random downtime or network issues don't affect the status for the whole next 10 minutes.

Regarding dashboard and Prometheus metrics, this could be solved by adding another state. For example, Uptime Kuma has a *pending* state that starts when the first check has failed, and changes after the configurable amount of attempts to either back *up* or *down*.
